src/include/crowbar/complex.rb

```diff
@@ -510,15 +607,144 @@ def HandleReposTable(key, event)
       selected = Convert.to_string(UI.QueryWidget(Id(key), :Value))
       if selected != nil && selected != @current_repo
         @current_repo = selected
+        @current_repo_platform = Convert.to_string(
+          UI.QueryWidget(Id(key), Cell(selected, 3))
+        )
+        InitRepoURL("repo_url")
+        InitAskOnError("ask_on_error")
+        InitTargetPlatform("target_platform")
+      end
+      nil
+    end
+
+    # handler for adding new repository button
+    def HandleAddRepositoryButton(key, event)
```

Comment:
NP: I'd split this very long method into smaller parts (like display the dialog, handle the user  input, activate the settings).